### ISSUE TITLE
Determine actual required Python Version

### ISSUE THREAD

**Issue by tjdevries** [sprout42/StarStruct#34]
Also I'm wondering how much work it would be to throw in compatibility for python 2.7.

I haven't really done anything for cross compatibility before.

We could use `tox` to help us with multiple test version.

**Comment by sprout42** [comment]
Good question...  I don't think that the current design should prevent us from supporting python 2.7.  For future use we may need to check the current python version and use a standard dictionary rather than OrderedDict.  But aside from that and adding some `from __future__ import print` statements... it should be pretty good?

**Comment by tjdevries** [comment]
I think the type hinting won't work. We could try and use stub files I
suppose... But I like the way it looks right now.

I think we can support 3.4 though.

**Comment by sprout42** [comment]
mypy (http://mypy-lang.org/) does have support type hints for python2.7, although they aren't as nice as the newer syntax.

**Comment by tjdevries** [comment]
We now use byte strings and regular strings differently, so I think python 2.x is out.

**Comment by sprout42** [comment]
true, and I don't really feel like supporting the older versions anyway, we should just do some py3 tests to see how old it supports and call it good.
